src/main/java/hlf/java/rest/client/service/impl/TransactionFulfillmentImpl.java

```diff
@@ -466,4 +470,96 @@ public ResponseEntity<EventAPIResponseModel> readTransactionEventByBlockNumber(
       throw new ServiceException(ErrorCode.HYPERLEDGER_FABRIC_CHANNEL_TXN_ERROR, e.getMessage(), e);
     }
   }
+
+  @Override
+  public ResponseEntity<ClientResponseModel> writeMultiDataTransactionToLedger(
+      String networkName,
+      String contractName,
+      String transactionFunctionName,
+      MultiDataTransactionPayload multiDataTransactionPayload) {
+
+    // Validate the incoming payload
+    AbstractModelValidator<MultiDataTransactionPayload> validator =
+        new MultiPrivateDataTransactionPayloadValidator();
+    validator.validate(multiDataTransactionPayload);
+
+    Collection<Peer> endorsingPeers = new ArrayList<>();
+    String resultString;
+
+    try {
+
+      Network network = gateway.getNetwork(networkName);
+      Contract contract = network.getContract(contractName);
+      Transaction fabricTransaction = contract.createTransaction(transactionFunctionName);
+      // override default commithandler to wait for any response from msp
+      fabricTransaction.setCommitHandler(DefaultCommitHandlers.MSPID_SCOPE_ANYFORTX);
+
+      endorsingPeers =
+          network.getChannel().getPeers().stream()
+              .filter(peer -> multiDataTransactionPayload.getPeerNames().contains(peer.getName()))
+              .collect(Collectors.toList());
+      if (!endorsingPeers.isEmpty()) {
+        fabricTransaction.setEndorsingPeers(endorsingPeers);
+      } else {
+        log.warn("Peer names don't match channel peers");
```

Comment:
Since peers info is optional, so shouldn't `channel.getPeers` be called here, to get peers if no peer info is passed

Reply:
Population of endorserPeer list that is seen here is similar to how it's been implemented in other `write` methods, where endorser peer List is constructed based on the matching incoming Peer names with Channel Peers, if incoming Peer-List is empty or null, `setEndorsingPeers`  is not called.